snapcraft/internal/lifecycle.py

```diff
@@ -243,6 +243,20 @@ def _run_step(self, step, part, part_names):
         common.env.extend(self.config.project_env())
 
         part = _replace_in_part(part)
+
+        # Add the annotated list of build packages
+        build_packages = repo.Repo.get_installed_build_packages(
```

Comment:
My only concern with having this here is that it'll get re-calculated for every step of the same part, which will slow things down. Is there a place where we can calculate it only once? Perhaps where the part is created, or something similar? What do you think?